src/main/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWinters.scala

```diff
@@ -0,0 +1,233 @@
+  *
+  * @param seasonality: Cycle length of the metric
+  * @param metricsInterval: How often a metric is available
+  */
+class HoltWinters(
+    seasonality: HoltWinters.SeriesSeasonality.Value,
+    metricsInterval: HoltWinters.MetricInterval.Value)
+  extends AnomalyDetectionStrategy {
+
+  import HoltWinters._
+
+  private val seriesPeriodicity = seasonality -> metricsInterval match {
+    case (SeriesSeasonality.Weekly, MetricInterval.Daily) => 7
+    case (SeriesSeasonality.Yearly, MetricInterval.Monthly) => 12
+  }
+
+  /**
+    * Triple exponential smoothing with additive trend and seasonality
+    * ETS(A, A) according to https://otexts.org/fpp2/taxonomy.html
+    *
+    * References:
+    * - https://otexts.org/fpp2/holt-winters.html
+    * - https://gist.github.com/andrequeiroz/5888967
+    *
+    * @param series: input time series
+    * @param periodicity: periodicity of series
+    * @param numberOfPointsToForecast: number of data points to forecast
+    * @return (forecasts, one step ahead residual SD)
+    */
+  private[seasonal] def additiveHoltWinters(
+      series: Seq[Double],
+      periodicity: Int,
+      numberOfPointsToForecast: Int,
+      alpha: Double,
+      beta: Double,
+      gamma: Double)
+    : ModelResults = {
+
+    val firstPeriodSum = series.take(periodicity).sum
+    val secondPeriodSum = series.slice(periodicity, 2 * periodicity).sum
+
+    // mean of first period
+    val level = ListBuffer(firstPeriodSum / periodicity.toDouble)
+
+    // mean of second period - mean of first period
+    val trend = ListBuffer(
+      (secondPeriodSum - firstPeriodSum) / (periodicity.toDouble * periodicity.toDouble)
+    )
+
+    // first `periodicity` data points - estimated level
+    val seasonality = ListBuffer(series.take(periodicity).map(_ - level.head): _*)
+    // running signal estimate
+    val y = ListBuffer(level.head + trend.head + seasonality.head)
+    // input `series`, `numberOfPointsToForecast` forecasts will be appended here
+    val Y = ListBuffer(series: _*)
+
+    for (t <- 0 until series.size + numberOfPointsToForecast) {
+      if (t >= series.size) {
+        Y.append(level.last + trend.last + seasonality(seasonality.size - periodicity))
+      }
+
+      level.append(alpha * (Y(t) - seasonality(t)) + (1 - alpha) * (level(t) + trend(t)))
+      trend.append(beta * (level(t + 1) - level(t)) + (1 - beta) * trend(t))
+      seasonality.append(gamma * (Y(t) - level(t) - trend(t)) + (1 - gamma) * seasonality(t))
+
+      y.append(level(t + 1) + trend(t + 1) + seasonality(t + 1))
+    }
+
+    // forecast residuals
+    val residuals = y.zip(series).map { case (modelForecast, seriesValue) =>
+      seriesValue - modelForecast
+    }
+    val forecasted = Y.drop(series.size)
+
+    ModelResults(forecasted, level, trend, seasonality, residuals)
+  }
+
+  private def modelSelectionFor(
+      dataSeries: Seq[Double],
+      numberOfObservationsToForecast: Int)
+    : (Double, Double, Double) = {
```

Comment:
can you create a case class for the return type? then users don't need to guess what the tuple fields stand for